packages/techdocs-common/src/stages/publish/awsS3.ts

```diff
@@ -135,25 +134,25 @@ export class AwsS3Publish implements PublisherBase {
       return await new Promise<string>((resolve, reject) => {
         const entityRootDir = `${entityName.namespace}/${entityName.kind}/${entityName.name}`;
 
-        const fileStreamChunks: Array<any> = [];
         this.storageClient
           .getObject({
             Bucket: this.bucketName,
             Key: `${entityRootDir}/techdocs_metadata.json`,
           })
-          .createReadStream()
-          .on('error', err => {
+          .then(file => {
+            const techdocsMetadataJson = file?.Body?.toString();
```

Comment:
Hey! Unfortunately this might need something else here. Could you try opening some docs site locally? I am receiving `[object Object]` as response (for the Metadata API and for the request of `index.html` using `docsRouter()` below.

Reply:
I'll look at it more closely !

At the time of the implementation I had some questions regarding the `getObject` method https://github.com/backstage/backstage/pull/3794#discussion_r547181420.

I'm going to fix that and make some tests locally to make sure everything works 👍

Reply:
Aah sorry, I missed that question in the previous PR!